server/core/terraform/terraform_client.go

```diff
@@ -128,17 +119,19 @@ func NewClientWithDefaultVersion(
 	versions := make(map[string]string)
 	var versionsLock sync.Mutex
 
-	localPath, err := exec.LookPath("terraform")
+	localPath, err := exec.LookPath(distribution.BinName())
 	if err != nil && defaultVersionStr == "" {
-		return nil, fmt.Errorf("terraform not found in $PATH. Set --%s or download terraform from https://developer.hashicorp.com/terraform/downloads", defaultVersionFlagName)
+		return nil, fmt.Errorf("%s not found in $PATH. Set --%s or download terraform from https://developer.hashicorp.com/terraform/downloads", distribution.BinName(), defaultVersionFlagName)
 	}
 	if err == nil {
 		localVersion, err = getVersion(localPath)
 		if err != nil {
+			panic(err)
```

Comment:
Do we want to panic here? I do not want to introduce new panics to the codebase without good reason.